### ISSUE TITLE
WaitForFixedUpdate does not work as described

### ISSUE THREAD

**Issue by Qriva** [Cysharp/UniTask#377]
This line from documentation is wrong:
```
// replacement of yield return new WaitForFixedUpdate(same as UniTask.Yield(PlayerLoopTiming.FixedUpdate))
await UniTask.WaitForFixedUpdate();
```
Coroutine WaitForFixedUpdate is [called the same frame after physics update](https://docs.unity3d.com/Manual/ExecutionOrder.html#ScriptLifecycleFlowchart) and when used in OnTirggerXYZ family callbacks it will correctly be called the same frame after callbacks. Unitask behaviour is different - it will be called next frame and probably at the beggining of FixedUpdate. To get  the same behaviour I needed to use `UniTask.Yield(PlayerLoopTiming.LastFixedUpdate);`

**Comment by neuecc** [comment]
Thank you very much.
After verification, I may change the implementation of WaitForFixedUpdate (since the same behavior is desirable).

**Comment by aidenatgravitysketch** [comment]
Please. I just came across the same issue.

**Comment by neuecc** [comment]
released at 2.3.2.
